Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -4122,7 +4122,8 @@ internal void RestoreStateWithoutDevices(SerializedState state)
 
             if (m_Settings != null)
                 Object.DestroyImmediate(m_Settings);
-            m_Settings = state.settings;
+                
+            settings = state.settings;
```

Comment:
So the theory here is that using the property instead of a direct setter makes sure we update all the derived data (static caches) -- that happens in ApplySettings that is ignored in here.